### PR TITLE
[PY] fix: Map tokens config to `max_tokens` when non-o1 model is used.

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #minor

## Details
Using the `max_completion_tokens` field is not supported for non-o1 models in Azure OpenAI.

#### Change details
* In `openai_model.py` if the model is not in the o1 series, then use `max_tokens` field by default.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by corinagum — COMMENTED** [review]
unit tests?

**Review by lilyydu — APPROVED** [review]

**Review by corinagum — APPROVED** [review]

File: python/packages/ai/teams/ai/models/openai_model.py [hunk 1]
```diff
@@ -202,77 +202,15 @@ async def complete_prompt(
...
-                        role="system",
-                        content=msg.content if msg.content is not None else "",
-                    )
-
-                if msg.name:
-                    param["name"] = msg.name
-
-            messages.append(param)
+        messages: List[chat.ChatCompletionMessageParam]
+        messages = self._map_messages(res.output, is_o1_model)
```
Comment (lilyydu) [inline python/packages/ai/teams/ai/models/openai_model.py:206]: could we merge these lines together?

**Review by lilyydu — COMMENTED** [review]
Thanks for adding tests, just 1 nit

File: python/packages/ai/teams/ai/models/openai_model.py (hunk 1 above)
Comment (singhk97) [reply python/packages/ai/teams/ai/models/openai_model.py:206]: that would make the line too long and throw a lint error

**Review by lilyydu — APPROVED** [review]

**Pull request merged**